### ISSUE TITLE
Hide actual cursor when capturing cursor

### ISSUE THREAD

**Issue by somatronicon** [rohatsu/ShaderGlass#10]
I'm using this full-screen on my main machine because it's awesome. One thing I noticed is that although I have the cursor captured, I'm still seeing my actual cursor ghosting around in front of it. When using shaders with a curve, the cursor is visible as I near the corners and is offset due to the curvature.

I think there is a Win32 API to hide the cursor in a Window, I just haven't been down that road for so long I don't remember what it is.

BTW, +1 for using C++ and Win32 API; that's how I cut my teeth on C++ and MASM32 20+ years ago.

My preset is as follows:

```
ProfileVersion "1.0"
PixelSize "1"
AspectRatio "1"
ShaderCategory "crt"
ShaderName "crt-lottes-multipass"
FrameSkip "2"
OutputScale "100"
FlipH "0"
FlipV "0"
Clone "0"
CaptureCursor "1"
Transparent "1"
```

**Comment by rohatsu** [comment]
Hey apologies for the late reply, the problem is if I hide the cursor then it completely disappears, including from the capture... not seeing a good way around this other than adding a cursor manually onto the captured image (bit of a can of worms). The main use-case for Capture Cursor was with Clone mode although that probably won't work for you in full-screen.

**Comment by RationalFragile** [comment]
Dealing with the same issue while trying to flip my screen and having two cursors.

I have an idea:
Magpie is able to somehow able to transform the cursor to the correct shape even though it is physically not in that place. Example, if you select an explorer window and apply Magpie to it, and then hover over a text field, the cursor changes to text-edit shape, even though the location of the Windows' cursor is different than the location of the text field in the non-scaled window.
So, a solution to this issue in ShaderGlass can be to hide the Windows's cursor, capture all mouse input with a hook, pass along the mouse to the desktop, and then get the shape of the cursor, apply it to the input, and so it will get drawn correctly in the output.

**Comment by skygate2012** [comment]
@RationalFragile Exactly. I wrote an AHK script earlier that's implemented this way. Would love to see this implemented natively.

**Comment by RationalFragile** [comment]
@skygate2012 Sounds cool! Can you share your AHK script?

**Comment by skygate2012** [comment]
@RationalFragile Sure
[mouse_emulator.zip](https://github.com/mausimus/ShaderGlass/files/8494598/mouse_emulator.zip)
!

**Comment by TakahashiEdenyako** [comment]
How do you open mouse_emulator.zip with shader glass? I want my cursor to match with the shader.

**Comment by DavidEscalante** [comment]
I'm not sure how useful this is, but I discovered that when I disable the hardware cursor in StarCraft 1, it hides the Windows cursor, and you only see the cursor displayed on the shader glass screen.

**Comment by mausimus** [comment]
I've been working on hiding the "unshaded" cursor, should be in the next beta in a few weeks.

**Comment by mausimus** [comment]
I released 1.1.5 beta which automatically hides original cursor when "Capture Cursor" is enabled, please give it a try.
https://github.com/mausimus/ShaderGlass/releases
